cmake/hipsycl-config.cmake.in

```diff
@@ -48,40 +48,66 @@ if(HIPSYCL_ROCM_BACKEND_AVAILABLE)
   endif()
 endif()
 
+set(HIPSYCL_SYCLCC_EXTRA_ARGS "${HIPSYCL_SYCLCC_EXTRA_ARGS}" CACHE STRING "Arguments to pass through directly to syclcc.")
+
+set(HIPSYCL_TARGETS "${HIPSYCL_TARGETS}" CACHE STRING "The platforms and architectures hipSYCL should target, in the format <platform1>[:arch1[,arch2][..,archN]][;<platform2>[:arch1][...]][..;<platformN>] where platforms are one of ${HIPSYCL_PLATFORMS_STRING}.")
+# IF HIPSYCL_TARGETS has not been explicitly set by the user, first tro to find the corresponding environment variable.
+# If found, takes precedence over deprecated HIPSYCL_PLATFORM and HIPSYCL_GPU_ARCH
+# IF not found, fallback to deprecated HIPSYCL_PLATFORM and HIPSYCL_GPU_ARCH logic
+if(NOT HIPSYCL_TARGETS)
+  set(HIPYSCL_TARGETS_ENV $ENV{HIPSYCL_TARGETS})
+  if(HIPSYCL_TARGETS_ENV)
+    message("Found HIPYSCL_TARGETS from environment: ${HIPSYCL_TARGETS_ENV}")
+    set(HIPSYCL_TARGETS "${HIPSYCL_TARGETS_ENV}")
+  endif()
+endif()
+if(HIPSYCL_TARGETS)
+  # Canonicalize all targets
+  set (HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS}")
+  string(REGEX REPLACE "cpu|host|hipcpu|openmp|omp" "omp" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
+  string(REGEX REPLACE "cuda|nvidia" "cuda" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
+  string(REGEX REPLACE "rocm|amd|hip|hcc" "hip" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
```

Comment:
As I've said previously, it's not clear to me why we need this here - `syclcc --hipsycl-targets` does not support the old synonyms, and I think we should align what cmake supports with what `syclcc` supports. That way, we can deprecate and remove the synonyms in the future altogether.

Reply:
Thanks for the reminder! Had other fires to put out this week, but I'll try to push a commit stripping this canonicalization out before the end of the day

Reply:
Fixed in 2fa9df2